dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,329 @@
+
+        System.setProperty("es.index.number_of_shards", String.valueOf(shards));
+        Logger.info(this, "Running Contentlet Reindex");
+
+        if(!DOTALL.equals(contentType)) {
+            ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(contentType);
+            Logger.info(this.getClass(), "Starting reindex of " + type.name());
+            APILocator.getContentletIndexAPI().removeContentFromIndexByStructureInode(type.id());
+            APILocator.getContentletAPI().refresh(type);
+        }
+        else {
+            APILocator.getContentletAPI().refreshAllContent();
+        }
+
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+
+
+    }
+
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/reindex")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response stopReindexation(@Context final HttpServletRequest request, 
+                    @Context final HttpServletResponse response, 
+                    @DefaultValue("true") @QueryParam("switch") boolean switchMe)
+                    throws DotDataException {
+        validateUser(request, response);
+        
+        if(!APILocator.getContentletIndexAPI().isInFullReindex()) {
+            return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+        }
+        if(switchMe) {
+            APILocator.getContentletIndexAPI().stopFullReindexationAndSwitchover();
+        }else {
+            
+            APILocator.getContentletIndexAPI().stopFullReindexation();
+        }
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+    }
+
+
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/clean")
+    public Response cleanContentType(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    String inode) throws DotDataException, DotSecurityException, LanguageException {
+
+        final InitDataObject init = validateUser(request, response);
+        ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(inode);
+        APILocator.getContentletIndexAPI().removeContentFromIndexByStructureInode(type.id());
+        APILocator.getContentletAPI().refresh(type);
+
+        String message = LanguageUtil.get("message.cmsmaintenance.cache.indexrebuilt", init.getUser());
+
+
+
+        return Response.ok(new ResponseEntityView(message)).build();
+
+
+    }
+
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/optimize")
+    public Response optimizeIndices(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        validateUser(request, response);
+        ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        List<String> indices = api.listDotCMSIndices();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'indices' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)